flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -186,7 +186,12 @@ object ScalarOperatorGens {
 
       case (TIME_WITHOUT_TIME_ZONE, INTERVAL_DAY_TIME) =>
         generateOperatorIfNotNull(ctx, new TimeType(), left, right) {
-          (l, r) => s"$l $op ((int) ($r))"
+          (l, r) => s"java.lang.Math.toIntExact(($l $op $r) % ${MILLIS_PER_DAY}L)"
```

Comment:
should be 
```
(l, r) => s"$l $op (java.lang.Math.toIntExact($r % ${MILLIS_PER_DAY}L)))"
```
to make it consistent with the table-planner one?

Reply:
> should be
> 
> ```
> (l, r) => s"$l $op (java.lang.Math.toIntExact($r % ${MILLIS_PER_DAY}L)))"
> ```
> 
> to make it consistent with the table-planner one?

Let me change it.

Reply:
> (l, r) => s"$l $op (java.lang.Math.toIntExact($r % ${MILLIS_PER_DAY}L)))"

It doesn't seem to change to this, it will report the following error:
```scala
result$3 = ((int) 86399000) + (java.lang.Math.toIntExact (result$1% 86400000L)) "
is not a type
 ```

Reply:
hmm it might be some typo problem here. I will fix it and merge it.